panoptes_client/workflow.py

```diff
@@ -530,6 +532,55 @@ def configure_for_alice(self):
         self.add_alice_reducers()
         self.add_alice_rules_and_effects()
 
+    def get_batch_aggregations(self):
+        """
+        This method will fetch existing aggregations if any. It will return a dict with the existing aggregations.
+        """
+
+        return BatchAggregation().get_aggregations(self.id)[0]
+
+    def run_batch_aggregation(self, user=None, delete_if_exists=False):
+        """
+        This method will start a new batch aggregation run, Will return a dict with the created aggregation if successful.
+
+        - **user** can be either a :py:class:`.User` or an ID.
+        - **delete_if_exists** parameter is optional if true, deletes any previous instance
+        -
+        Examples::
+
+            Workflow(1234).run_batch_aggregation(1234)
+            Workflow(1234).run_batch_aggregation(user=1234, delete_if_exists=True)
+        """
+
+        if(isinstance(user, User)):
+            _user_id = user.id
+        elif (isinstance(user, (int, str,))):
+            _user_id = user
+        else:
+            raise TypeError
+
+        payload = {
+            "aggregations": {
+                "links": {
+                    "user": _user_id,
+                    "workflow": self.id,
+              }
+            }
+        }
+        return BatchAggregation().run_aggregation(payload, delete_if_exists)
+
+    def check_batch_aggregation_run_status(self):
+        """
+        This method will fetch existing aggregation status if any.
+        """
+        return self.get_batch_aggregations()['aggregations'][0]['status']
```

Comment:
I think there should be a check if there are no aggregations. i.e. 
when  `self.get_batch_aggregations()['aggregations']` length of resulting array is 0.  

Same with `get_batch_aggregation_links` (method below on L578)